### DISCUSSION TITLE
Google Search without SerpAPI

### DISCUSSION THREAD

**Discussion by CXwudi** [langgenius/dify#4420] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

I want to use Google Search API to perform the Google Search, not by using any 3rd party search engine like SerpAPI.

### 2. Describe the feature you'd like to see

Implement a tool that natively use Google Search API, like [LibreChat's Google Search plugin](https://github.com/danny-avila/LibreChat/blob/main/api/app/clients/tools/structured/GoogleSearch.js)

### 3. How will this feature improve your workflow or experience?

This feature will eliminate the limitation of only 100 free searches per month in free SerpAPI account.

### 4. Additional context or comments

See also https://www.librechat.ai/docs/configuration/tools/google_search

### 5. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by SoulGolf** [comment]
serper.dev 每个月有2500次的免费搜索次数,比SerpAPI好很多,建议替换成serperAPI
